### PR TITLE
Fix Output:Table:Annual references to schedules not showing name in column heading

### PR DESCRIPTION
**Pull request by JasonGlazer**
Addresses #5912 so schedules referenced in Output:Table:Annual show their names in the column heading and output variables and meters are converted to title case.

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

… real output variables and meters

### CODE REVIEW

**Comment by JasonGlazer** [comment]
CI is clean so this is ready to be reviewed and merged.

**Comment by nrel-bot-3** [comment]
@JasonGlazer @lgentile it has been 14 days since this pull request was last updated.

File: tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc [hunk 1]
```diff
@@ -219,3 +219,68 @@ TEST_F( EnergyPlusFixture, OutputReportTabularAnnual_GatherResults )
...
+		"    Until: 20:00, 1.0,       !- Field 6",
+		"    Until: 24:00, 0.0,       !- Field 8",
+		"    For: AllOtherDays,       !- Field 9",
+		"    Until: 24:00, 0.0;       !- Field 11",
+	});
+
+	ASSERT_FALSE (process_idf (idf_objects));	
+
+	Real64 facilUse;
+	SetupOutputVariable ("Misc Facility Electric Energy [J]", facilUse, "Zone", "Sum", "Lite1", _, "Electricity", "Facility", "General"); //create an electric meter
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc]: Padding around parentheses is sorta backward from our usual "style"...I may push up restyled code.

File: tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc [hunk 2]
```diff
@@ -219,3 +219,68 @@ TEST_F( EnergyPlusFixture, OutputReportTabularAnnual_GatherResults )
...
+	OutputProcessor::NumEnergyMeters = 2;
+	OutputProcessor::EnergyMeters.allocate(OutputProcessor::NumEnergyMeters);
+	OutputProcessor::EnergyMeters(1).Name = "Electricity:Facility"; //"ELECTRICITY:FACILITY";
+	OutputProcessor::EnergyMeters(2).Name = "ELECTRICITY:LIGHTING";
+
+	DataGlobals::DoWeathSim = true;
+
+	OutputReportTabularAnnual::GetInputTabularAnnual();
+
+	EXPECT_EQ (OutputReportTabularAnnual::annualTables.size (), 1u);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc]: Another odd spacing here.  Maybe just double check that visual studio has the right (or close enough) auto style settings.

File: tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc [hunk 3]
```diff
@@ -219,3 +219,68 @@ TEST_F( EnergyPlusFixture, OutputReportTabularAnnual_GatherResults )
...
+
+	std::vector<AnnualTable>::iterator firstTable = OutputReportTabularAnnual::annualTables.begin ();
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc]: Is there a reason to not just get the first table? Doesn't look like you are doing anything below that requires an iterator.

`AnnualTable & firstTable = OutputReportTabularAnnual::annualTables[0];`

File: src/EnergyPlus/OutputReportTabularAnnual.cc [hunk 4]
```diff
@@ -1363,8 +1363,12 @@ namespace EnergyPlus {
 			for ( fldStIt = m_annualFields.begin(); fldStIt != m_annualFields.end(); ++fldStIt ){
 				if ( fldStIt->m_variMeter == fldStIt->m_colHead ){
 					if ( fldStIt->m_indexesForKeyVar.size() > 0 ){
-						int varNum = fldStIt->m_indexesForKeyVar[0];
-						fldStIt->m_colHead = OutputProcessor::RVariableTypes[varNum].VarNameOnly;
+						int varNum = fldStIt->m_indexesForKeyVar[ 0 ];
+						if (fldStIt->m_typeOfVar == OutputProcessor::VarType_Real ) {
+							fldStIt->m_colHead = OutputProcessor::RVariableTypes(varNum).VarNameOnly;
+						} else if ( fldStIt->m_typeOfVar == OutputProcessor::VarType_Meter ) {
+							fldStIt->m_colHead = OutputProcessor::EnergyMeters(varNum).Name;
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReportTabularAnnual.cc:1358]: Looks good.

**Review by Myoldmopar — CHANGES_REQUESTED** [review]
@JasonGlazer Can you just comment on the unit test question below regarding whether you actually needed an iterator?  I'll clean up the spaces in the code and if the iterator isn't needed I'll make that a bit simpler.

File: tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc (hunk 3 above)
Comment (JasonGlazer) [reply tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc]: No iterating. Seemed more c++ish  to me.

**Comment by nrel-bot-3** [comment]
@JasonGlazer @lgentile it has been 15 days since this pull request was last updated.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Built, played with locally.  Fixed whitespace.  Left the iterator stuff in there, it's fine.  Merging.

**Pull request merged**